### PR TITLE
Update custom server examples

### PR DESCRIPTION
**Pull request by timneutkens**
Fixes the links/readme descriptions, also removes legacy routing that can now be handled using `rewrites` which also work client-side automatically.

Closes #24607

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.

## Documentation / Examples

- [ ] Make sure the linting passes

### CODE REVIEW

File: examples/custom-server-hapi/server.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const next = require('next')
 const Hapi = require('@hapi/hapi')
-const { pathWrapper, nextHandlerWrapper } = require('./next-wrapper')
+const { /* pathWrapper, */ nextHandlerWrapper } = require('./next-wrapper')
```
Comment (leerob) [inline examples/custom-server-hapi/server.js:3]:
```suggestion
const { nextHandlerWrapper } = require('./next-wrapper')
```

Should this be removed?

File: examples/custom-server-express/server.js [hunk 2]
```diff
@@ -9,14 +9,6 @@ const handle = app.getRequestHandler()
...
-    return app.render(req, res, '/a', req.query)
-  })
-
-  server.get('/b', (req, res) => {
-    return app.render(req, res, '/b', req.query)
-  })
-
   server.all('*', (req, res) => {
     return handle(req, res)
   })
```
Comment (cseas) [inline examples/custom-server-express/server.js:14]: As someone trying to understand how the Next.js custom Express setup works, I have some questions:

If the `server.get()` path handlers are not needed here, does that mean the `server.all()` handler will just know how to handle all the paths defined in [pages](https://github.com/vercel/next.js/blob/243472f34d58e516c8980eadccb276ff85611f9f/examples/custom-server-express/pages/index.js#L7-L12)? So I'm assuming we can just define API endpoints here that Express needs to handle and all the application path routing will just work.

If yes, then what's the point of all the `app.render` in the [documentation example](https://nextjs.org/docs/advanced-features/custom-server)? Shouldn't that only be done when file-system routing is disabled? Having it in the basic example makes it look like it's necessary to define all the path handlers.

I'm also confused about what the `query` parameter is and how and when it is supposed to be used. In the lines deleted here, we were using `req.query` and that worked fine. But the documentation example shows that `query` should be taken from the result of `parse(req.url, true)` and then passed to the `app.render`. Can't find any explanation of what `query` parameter is exactly supposed to be and why does `app.render` need it.

File: examples/custom-server-hapi/server.js (hunk 1 above)
Comment (timneutkens) [reply examples/custom-server-hapi/server.js:3]: No, I left it intentionally.

File: examples/custom-server-express/server.js (hunk 2 above)
Comment (cseas) [reply examples/custom-server-express/server.js:14]: For context, my use case is that we have a fat Express server serving the `build` folder from `create-react-app`. Trying to figure out how to migrate to Next.js to do server-rendering.

I'd expect the documentation/examples to have a mention of:
- Where do the express middlewares fit here?
- Is it necessary to define path handlers for every route of the application?
- Where are the response headers supposed to be in a custom server with Next.js? Will express keep handling the headers for every page request or will Next.js headers override that functionality? Are there any conflicts between the two?

Understandably, these questions could be directed to Discussions or StackOverflow but it'd be great if it's possible to figure these answers out from the documentation.

**Pull request merged**